retry/lib/retry.dart

```diff
@@ -177,8 +177,8 @@ Future<T> retry<T>(
   double randomizationFactor = 0.25,
   Duration maxDelay = const Duration(seconds: 30),
   int maxAttempts = 8,
-  FutureOr<bool> Function(Exception)? retryIf,
-  FutureOr<void> Function(Exception)? onRetry,
+  FutureOr<bool> Function(Exception, StackTrace)? retryIf,
```

Comment:
I don't think users should be allowed to condition the retry on the stacktrace.

if you're doing this something is very wrong. It's probably better to fix the thing that is wrong.

Reply:
I think stackTrace can provide additional information about an exception, which can help developers pinpoint the specific cause of the exception. This is especially helpful when exceptions occur sporadically, or when the location of the exception differs each time you retry. The stack trace can assist developers in reproducing and fixing these difficult to diagnose exceptions.

Reply:
Yes, retryIf should not carry the stackTrace parameter.
Updated.